packages/network-controller/src/NetworkController.ts

```diff
@@ -186,26 +295,35 @@ export type CustomNetworkClientId = string;
 export type NetworkClientId = BuiltInNetworkClientId | CustomNetworkClientId;
 
 /**
- * Information about networks not held by any other part of state.
+ * Extra information about each network, such as whether it is accessible or
+ * blocked and whether it supports EIP-1559, keyed by network client ID.
  */
-export type NetworksMetadata = {
-  [networkClientId: NetworkClientId]: NetworkMetadata;
-};
+export type NetworksMetadata = Record<NetworkClientId, NetworkMetadata>;
 
```

Comment:
nit: This is probably outside of the PR's scope, but `NetworkProperties` might be a possible replacement for `NetworkMetadata`, which could be confused with `NetworkStateMetadata`.

In general, we might want to look into categorizing our data types and properties so that they consistently distinguish between the following categories (and possibly more). `TransactionMeta`, for example, might benefit from being broken up into a few types:
- Static descriptors/metadata (e.g. `NetworkClientId`, `NetworkConfiguration['nativeTokenName']`)
- Transient state (`NetworkMetadata['status']`)
- Inherent/immutable properties that affect behavior (e.g. `NetworkMetadata['EIPS']`)
- Configurable/mutable properties that affect behavior (e.g. `NetworkConfiguration['rpcEndpoints]`)

Reply:
Good idea to rename `NetworkMetadata`. I have never quite liked that name.

Also a good idea to think about how state properties ought to be categorized. Knowing that some state properties are transient and can easily be recreated, and that some properties are expected to be immutable is important.